pkg/deploy/api/types.go

```diff
@@ -118,6 +123,14 @@ type ExecNewPodHook struct {
 	Volumes []string
 }
 
+// TagImageHook is a request to tag the image in a particular container onto an ImageStreamTag.
+type TagImageHook struct {
+	// ContainerName is the name of a container in the deployment config whose image value will be used as the source of the tag
+	ContainerName string
```

Comment:
This is defaultable when we only have one container.  I don't see that happening and it should be mentioned in the godoc.